### PR TITLE
async-profiler: init at 1.8.4

### PR DESCRIPTION
**Pull request by mschuwalow**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 115909` at 0a8370bb run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package still failed to build (pre-existing failures on hydra):</summary>
  <ul>
    <li><a href="https://gist.github.com/32c03c7f153a8d3f6109935329f57ec9">async-profiler</a></li></li>
  </ul>
</details>

File: pkgs/development/tools/async-profiler/default.nix [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  };
+
+  buildInputs = [ jdk8 ];
+
+  installPhase = ''
+    runHook preInstall
+    install -D "$src/profiler.sh" "$out/bin/async-profiler"
+    install -D build/jattach "$out/bin/jattach"
+    install -D build/libasyncProfiler.so "$out/lib/libasyncProfiler.so"
+    install -D -t "$out/share/java/" build/*.jar
```
Comment (SuperSandro2000) [inline pkgs/development/tools/async-profiler/default.nix:21]: Why are you coping the files back to build?

File: pkgs/development/tools/async-profiler/default.nix [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+  meta = with lib; {
+    description = "A low overhead sampling profiler for Java that does not suffer from Safepoint bias problem";
+    homepage    = "https://github.com/jvm-profiling-tools/async-profiler";
+    license     = licenses.asl20;
+    maintainers = with maintainers; [ mschuwalow ];
+    platforms   = with platforms; all;
+  };
+
+}
```
Comment (SuperSandro2000) [inline pkgs/development/tools/async-profiler/default.nix:41]:
```suggestion
  };
}
```

File: pkgs/development/tools/async-profiler/default.nix [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+      --replace 'PROFILER=$SCRIPT_DIR/build/libasyncProfiler.so' \
+                'PROFILER=${placeholder "out"}/lib/libasyncProfiler.so'
+  '';
+
+  meta = with lib; {
+    description = "A low overhead sampling profiler for Java that does not suffer from Safepoint bias problem";
+    homepage    = "https://github.com/jvm-profiling-tools/async-profiler";
+    license     = licenses.asl20;
+    maintainers = with maintainers; [ mschuwalow ];
+    platforms   = with platforms; all;
```
Comment (SuperSandro2000) [inline pkgs/development/tools/async-profiler/default.nix:38]:
```suggestion
    platforms   = platforms.all;
```

File: pkgs/top-level/all-packages.nix [hunk 4]
```diff
@@ -1000,6 +1000,8 @@ in
 
   async = callPackage ../development/tools/async {};
 
+  async-profiler = callPackage ../development/tools/async-profiler {};
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:1003]:
```suggestion
  async-profiler = callPackage ../development/tools/async-profiler { };
```

File: pkgs/development/tools/async-profiler/default.nix (hunk 1 above)
Comment (mschuwalow) [reply pkgs/development/tools/async-profiler/default.nix:21]: They are being copied from build to $out/share/java

**Comment by mschuwalow** [comment]
What still needs to be done here?

File: pkgs/development/tools/async-profiler/default.nix (hunk 1 above)
Comment (SuperSandro2000) [reply pkgs/development/tools/async-profiler/default.nix:21]: derp.

**Pull request merged**